code/modules/clothing/gloves/miscellaneous.dm

```diff
@@ -183,3 +183,153 @@
+		var/mob/living/student = user
+		style.teach(student, 1)
+		ADD_TRAIT(student, TRAIT_NOGUNS, NEMESIS_TRAIT)
+
+/obj/item/clothing/gloves/rapid/nemesis/dropped(mob/user)
+	. = ..()
+
+	if(!ishuman(user))
+		return
+
+	var/mob/living/owner = user
+	style.remove(owner)
+	REMOVE_TRAIT(owner, TRAIT_NOGUNS, NEMESIS_TRAIT)
+
+	if(shield.loc == src)
+		return
+
+	if(shield.active)
+		shield.active = FALSE
+		shield.icon_state = "[shield.base_icon_state][shield.active]"
+		shield.force = initial(shield.force)
+		shield.w_class = WEIGHT_CLASS_TINY
+		playsound(user, 'sound/weapons/saberoff.ogg', 35, TRUE)
+		shield.slowdown = 0
+
+	playsound(loc, 'sound/mecha/mechmove03.ogg', 50, TRUE)
+	user.dropItemToGround(shield, TRUE)
+	shield.forceMove(src)
+
+/obj/item/clothing/gloves/rapid/nemesis/proc/lose_charge(amount_to_lose = 1)
+	if(ishuman(loc))
+		var/mob/living/carbon/human/owner = loc
+		if(istype(owner.get_item_by_slot(ITEM_SLOT_BELT), /obj/item/storage/belt/security/nemesis)) //Just in case you somehow get rid of the belt while using nemesis suit
+			var/obj/item/storage/belt/security/nemesis/belt = owner.get_item_by_slot(ITEM_SLOT_BELT)
+			for(var/tick = 1 to amount_to_lose)
+				if(belt.overcharge)
+					belt.overcharge--
+					amount_to_lose--
+				else
+					break
+	charge = max(0, charge - amount_to_lose)
+	update_charge()
+
+/obj/item/clothing/gloves/rapid/nemesis/proc/gain_charge(amount_to_gain = 1)
+	charge += amount_to_gain
+	if(ishuman(loc))
+		var/mob/living/carbon/human/owner = loc
+		if(charge > NEMESIS_MAX_CHARGE) //Overcharge will be turned into stamina damage and stored in the belt. Use your brain and gadgets, not gloves only
+
+			owner.apply_damage(5 * (charge - NEMESIS_MAX_CHARGE), STAMINA)
+
+			if(world.time > antispam + 10 SECONDS)
+				antispam = world.time
+				to_chat(owner, "<span class='userdanger'>OVERCHARGE DETECTED. Process to deplete the charge to avoid possible shocks.</span>")
+
+			if(istype(owner.get_item_by_slot(ITEM_SLOT_BELT), /obj/item/storage/belt/security/nemesis))
+				var/obj/item/storage/belt/security/nemesis/belt = owner.get_item_by_slot(ITEM_SLOT_BELT)
+				belt.overcharge += (charge - NEMESIS_MAX_CHARGE)
+
+			charge = NEMESIS_MAX_CHARGE
+	else
+		charge = max(charge, NEMESIS_MAX_CHARGE) //...how did we get here?
+	update_charge()
+
+/obj/item/clothing/gloves/rapid/nemesis/proc/update_charge()
+	if(!ishuman(loc))
+		return
+	var/mob/living/carbon/human/owner = loc
+	if(istype(owner.get_item_by_slot(ITEM_SLOT_OCLOTHING), /obj/item/clothing/suit/armor/vest/nemesis))
+		var/obj/item/clothing/suit/armor/vest/nemesis/suit = owner.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+		suit.update_charge(charge)
+
+	if(istype(owner.get_item_by_slot(ITEM_SLOT_BELT), /obj/item/storage/belt/security/nemesis))
+		var/obj/item/storage/belt/security/nemesis/belt = owner.get_item_by_slot(ITEM_SLOT_BELT)
+		belt.update_charge(charge)
+
+/obj/item/clothing/gloves/rapid/nemesis/attack_hand(mob/user, list/modifiers)
```

Comment:
I'd prefer it if this (and retracting the shield) was an action button, to make it easier to flick mid-combat and easier to discover the existence of